client/src/utils/uploadbox.js

```diff
@@ -245,13 +245,16 @@ export class UploadQueue {
                 }
             });
         }
+        // Returns last added file index.
+        return this.nextIndex - 1;
     }
 
-    // Remove file from queue
+    // Remove file from queue and file set by index
     remove(index) {
-        // Removes the item identified by index.
+        const file = this.queue.get(index);
+        const fileSetKey = file.name + file.size;
         // Returns true if an item was removed, and false otherwise.
-        return this.queue.delete(index);
+        return this.queue.delete(index) && this.fileSet.delete(fileSetKey);
```

Comment:
Is the return value used anywhere? If not I recommend to just execute both commands.

Reply:
It's not. I removed it.